e2e_test/batch/types/array_ty.slt.part

```diff
@@ -16,8 +16,17 @@ select ARRAY[null];
 ----
 {NULL}
 
-statement error
-select ARRAY[];
+# PostgreSQL requires a cast from user for this. But we treat it as varchar[] here.
+# Because the cast source need to be a valid subexpr.
```

Comment:
If we do want to match PostgreSQL's behavior, we need to remember the context when binding parent expr `cast` and pass it to child expr `array[]`, similar to how `insert values` passes column types to `bind_values` to enable assignment cast rather than implicit cast there.

But this makes `bind_expr` more complicated and adds little value. If `array[null]` can be `varchar[]`, `array[]` can be as well.

Reply:
Found an easier way because the left side (`array[]`) is a fixed ast. We are now aligned with PostgreSQL.